### DISCUSSION TITLE
Recommendations for generating OpenAPI with Lambda Annotations?

### DISCUSSION THREAD

**Discussion by dguisinger** [aws/aws-lambda-dotnet#1763] [Q&A] [answered]
I've been looking for a solution for creating an OpenAPI file automatically for my APIs.  I've generally hosted AspNet WebAPIs inside lambda just for the purpose that it will generate swagger/OpenAPI files and host the swagger UI.

Lately I've been starting to simplify my stack and move from WebAPI to Lambda Annotations, and would really like a way to automate OpenAPI generation.   I know I can register all my types with API Gateway but that would all have to be hand coded in my CDK project, and API Gateway doesn't allow public access to Export OpenAPI, so its useless IMO for a public API definition endpoint.

I'm assuming by this point someone else has solved for this.  Is there a nuget package or a best practice or something along those lines?

Ideally I'd like a combined OpenAPI definition for *ALL* API Gateways hosted within the custom domain, but I'd settle for one OpenAPI definition per API Gateway

**Comment by bhoradc** [comment]
Hi @dguisinger, 

Will check on this with the .NET SDK team and get back to you. 

Regards, 
Chaitanya

**Comment by bhoradc** [comment, marked as answer]
Hi @dguisinger, 

Thank you for reaching out. I have checked with the .NET SDK team and can confirm that there isn't a built-in solution right now for this purpose. But we would encourage you to submit a `Feature Request`. The more users requesting this functionality, the higher the chance it gets implemented in the future. 

Regards, 
Chaitanya

**Reply by dguisinger** [reply]
Following up on this thread in case it helps future readers.  I originally asked this because there wasn’t a good solution at the time.

Recently I’ve released an open-source project that generates OpenAPI specs from AWS Lambda Annotations in C#/.NET at build time, and also supports consolidating specs across multiple Lambda projects.

Project site: [https://lambdaopenapi.dev](https://lambdaopenapi.dev)
GitHub repo: [https://github.com/oproto/lambda-openapi](https://github.com/oproto/lambda-openapi)

Hopefully this helps others who run into the same gap.
